File: app/src/main/java/pl/pola_app/ui/fragment/HelpMessageDialog.java
package pl.pola_app.ui.fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;

import pl.pola_app.databinding.DialogHelpInfoBinding;

public class HelpMessageDialog extends DialogFragment {

    private DialogHelpInfoBinding binding;

    public OnWantHelpButtonClickListener onWantHelpButtonClickListener = OnWantHelpButtonClickListener.NULL;

    public static HelpMessageDialog newInstance() {
        return new HelpMessageDialog();
    }

    public void setOnWantHelpButtonClickListener(OnWantHelpButtonClickListener onWantHelpButtonClickListener) {
        this.onWantHelpButtonClickListener =
                onWantHelpButtonClickListener != null ? onWantHelpButtonClickListener : OnWantHelpButtonClickListener.NULL;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
            @Nullable Bundle savedInstanceState) {
        binding = DialogHelpInfoBinding.inflate(inflater, container, false);
        binding.helpDialogWantToHelpButton.setOnClickListener(this::onWantToHelpButtonClick);
        binding.helpDialogNextTimeButton.setOnClickListener(this::onNextTimeButtonClick);
        return binding.getRoot();
    }

    public void onWantToHelpButtonClick(View view) {
        onWantHelpButtonClickListener.onWantHelpButtonClick();
        dismiss();
    }

    public void onNextTimeButtonClick(View view) {
        dismiss();
    }

    public interface OnWantHelpButtonClickListener {

        OnWantHelpButtonClickListener NULL = () -> { /*Do nothing */ };

        void onWantHelpButtonClick();
    }
}

File: app/src/main/java/pl/pola_app/ui/fragment/HelpMessageDialog.kt
package pl.pola_app.ui.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.DialogFragment
import butterknife.ButterKnife
import butterknife.OnClick
import pl.pola_app.R

class HelpMessageDialog : DialogFragment() {
    private var onWantHelpButtonClickListener = OnWantHelpButtonClickListener.NULL
    fun setOnWantHelpButtonClickListener(onWantHelpButtonClickListener: OnWantHelpButtonClickListener?) {
        this.onWantHelpButtonClickListener =
            onWantHelpButtonClickListener ?: OnWantHelpButtonClickListener.NULL
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.dialog_help_info, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        ButterKnife.bind(this, view)
    }

    @OnClick(R.id.help_dialog_want_to_help_button)
    fun onWantToHelpButtonClick() {
        onWantHelpButtonClickListener.onWantHelpButtonClick()
        dismiss()
    }

    override fun onDestroy() {
        super.onDestroy()
    }

    @OnClick(R.id.help_dialog_next_time_button)
    fun onNextTimeButtonClick() {
        dismiss()
    }

    interface OnWantHelpButtonClickListener {
        fun onWantHelpButtonClick()

        companion object {
            val NULL: OnWantHelpButtonClickListener = object : OnWantHelpButtonClickListener{
                override fun onWantHelpButtonClick() {

                }

            }
        }
    }

    companion object {
        fun newInstance(): HelpMessageDialog {
            return HelpMessageDialog()
        }
    }
}